lib/hotel.rb

```diff
@@ -0,0 +1,96 @@
+require 'date'
+require 'pry'
+
+class Hotel
+  attr_reader :rooms, :reservations, :blocks
+  
+  def initialize(rooms: [], reservations: [], blocks: [])
+    @rooms = rooms
+    @reservations = reservations
+    @blocks = blocks
+    
+    20.times do |i|
+      rooms << Room.new(id:(i+1).to_i)
+    end
+  end
+  
+  # Wave 2: View available rooms for a given date range
+  def find_available_rooms(start_date:, end_date:)
```

Comment:
Why are you using kwargs here?

Reply:
Sandi Metz told me to... (section 3.2.3).